sound/soc/sof/intel/hda-loader.c

```diff
@@ -314,44 +314,49 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 	init_waitqueue_head(&sdev->boot_wait);
 	sdev->boot_complete = false;
 
-	/* try attempting fw boot a few times before giving up */
+	/* try attempting ROM init a few times before giving up */
 	for (i = 0; i < HDA_FW_BOOT_ATTEMPTS; i++) {
 		tag = cl_dsp_init(sdev, stripped_firmware.data,
 				  stripped_firmware.size);
 
-		if (tag <= 0) {
-			dev_err(sdev->dev, "error: Error code=0x%x: FW status=0x%x\n",
-				snd_sof_dsp_read(sdev, HDA_DSP_BAR,
-						 HDA_DSP_SRAM_REG_ROM_ERROR),
-				snd_sof_dsp_read(sdev, HDA_DSP_BAR,
-						 HDA_DSP_SRAM_REG_ROM_STATUS));
-			dev_err(sdev->dev, "error: iteration %d of Core En/ROM load fail:%d\n",
-				i, tag);
-			ret = tag;
-			continue;
-		}
+		/* no retry anymore at success */
+		if (tag > 0)
+			break;
 
-		/* at this point DSP ROM has been initialized and
-		 * should be ready for code loading and firmware boot
-		 */
-		ret = cl_copy_fw(sdev, tag);
-		if (ret < 0) {
-			dev_err(sdev->dev, "error: iteration %d of load fw failed err: %d\n",
-				i, ret);
-			continue;
-		}
+		dev_err(sdev->dev, "error: Error code=0x%x: FW status=0x%x\n",
+			snd_sof_dsp_read(sdev, HDA_DSP_BAR,
+					 HDA_DSP_SRAM_REG_ROM_ERROR),
+			snd_sof_dsp_read(sdev, HDA_DSP_BAR,
+					 HDA_DSP_SRAM_REG_ROM_STATUS));
+		dev_err(sdev->dev, "error: iteration %d of Core En/ROM load fail:%d\n",
+			i, tag);
+		ret = tag;
+	}
+
+	if (ret < 0) {
+		dev_err(sdev->dev, "error: load fw failed after %d attempts with err: %d\n",
+			HDA_FW_BOOT_ATTEMPTS, ret);
+
```

Comment:
Better to revise error message to "errror:  ROM init failed ...“ to align with the code change.

Reply:
makes sense.